### PR TITLE
[9.0][WIP] Migrated product multi image

### PR DESCRIPTION
**Pull request by SodexisTeam**

### CODE REVIEW

File: product_multi_image/README.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+#. If you change images from the *Images* tab, it changes in the template too.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/135/8.0
+
+Known issues / Roadmap
+======================
+
+* Provide proper migration scripts from module product_images from 7.0.
```
Comment (elicoidal) [inline product_multi_image/README.rst]: from 8.0?

File: product_multi_image/README.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+* Migrate to v8 api when https://github.com/odoo/odoo/issues/10799 gets fixed.
```
Comment (elicoidal) [inline product_multi_image/README.rst]: adapt content

File: product_multi_image/README.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  "one2many_v_id_62"``. This bug has not been fixed yet, but a workaround is to
+  save and edit again to sort images.
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/product-attribute/issues>`_. In case of trouble, please
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed `feedback
```
Comment (elicoidal) [inline product_multi_image/README.rst]: adapt to newest template

**Comment by elicoidal** [comment]
Can you check travis?

File: product_multi_image/README.rst (hunk 2 above)
Comment (pedrobaeza) [inline product_multi_image/README.rst]: This should be removed this time. It makes no sense.

**Comment by atchuthan** [comment]
@Yajo: I would like your help on the hook i.e. pre_init_hook(https://github.com/OCA/server-tools/blob/9.0/base_multi_image/hooks.py).

For version 9.0, it seems the attachment=True parameter for fields.Binary() makes the field unavailable in the current table and is stored in ir_attachments as a new record.

**Comment by pedrobaeza** [comment]
And what is the question or the problem?

**Comment by atchuthan** [comment]
When we try to call the hook available in base_multi_image with **pre_init_hook_for_submodules(cr, "product.template", "image")**, it results in ProgrammingError: column "image" does not exist.

So, I would like some suggestions on how this scenario could be handled.

cc: @Yajo

**Comment by yajo** [comment]
Interesting, I did not notice the new attachment feature. It's possible that now base module needs some refactoring to use that storage by default too.

**Comment by pedrobaeza** [comment]
Yeah, I also think that one is something to be taken into account in base_multi_image, not here.

**Comment by atchuthan** [comment]
@dreispt Why close this PR?  This PR uses the fix available in https://github.com/OCA/server-tools/pull/468.

Also, you might have confused this module(OCA/product-attribute/product_multi_image) with OCA/server-tools/base_multi_image.

**Comment by dreispt** [comment]
@atchuthan It was accidental, I don't what happened. Sorry! Reopened.

**Comment by atchuthan** [comment]
okay @dreispt

**Comment by yajo** [comment]
Seems like travis is still failing for some reason. Tests are not passing, there must be a bug.

**Comment by yajo** [comment]
This module received a major update in v8 in https://github.com/OCA/product-attribute/pull/151, could you merge with your migration to get that please?

**Comment by yajo** [comment]
Functional test in runbot fails:
1. Create a product.
2. Set an image from the thumbnail.

```
Odoo Server Error

Traceback (most recent call last):
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/http.py", line 319, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/service/model.py", line 118, in wrapper
    return f(dbname, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/http.py", line 312, in checked_call
    result = self.endpoint(*a, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/http.py", line 962, in __call__
    return self.method(*args, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/addons/web/controllers/main.py", line 898, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/addons/web/controllers/main.py", line 890, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/addons/product/product.py", line 675, in create
    product_template_id = super(product_template, self).create(cr, uid, vals, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 354, in old_api
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/addons/mail/models/mail_thread.py", line 233, in create
    thread = super(MailThread, self).create(values)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/models.py", line 4166, in create
    self._fields[key].determine_inverse(record)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/fields.py", line 954, in determine_inverse
    self.inverse(records)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/fields.py", line 595, in _inverse_related
    other[field.name] = record_value[record]
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/models.py", line 5761, in __setitem__
    return self._fields[key].__set__(self, value)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/fields.py", line 865, in __set__
    record.write({self.name: self.convert_to_write(value)})
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 574, in new_api
    result = method(self._model, cr, uid, self.ids, *args, **old_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/addons/product/product.py", line 699, in write
    res = super(product_template, self).write(cr, uid, ids, vals, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 381, in old_api
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/addons/mail/models/mail_thread.py", line 279, in write
    result = super(MailThread, self).write(values)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/models.py", line 3848, in write
    self._fields[key].determine_inverse(self)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/fields.py", line 952, in determine_inverse
    getattr(records, self.inverse)()
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/addons/base_multi_image/models/owner.py", line 81, in _set_multi_image_main
    self._set_multi_image(self.image_main)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/addons/base_multi_image/models/owner.py", line 74, in _set_multi_image
    s.image_ids = [(0, 0, values)]
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/fields.py", line 865, in __set__
    record.write({self.name: self.convert_to_write(value)})
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 574, in new_api
    result = method(self._model, cr, uid, self.ids, *args, **old_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/addons/product/product.py", line 699, in write
    res = super(product_template, self).write(cr, uid, ids, vals, context=context)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 381, in old_api
    result = method(recs, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/addons/mail/models/mail_thread.py", line 279, in write
    result = super(MailThread, self).write(values)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/models.py", line 3837, in write
    self._write(old_vals)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/api.py", line 574, in new_api
    result = method(self._model, cr, uid, self.ids, *args, **old_kwargs)
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/models.py", line 3995, in _write
    result += self._columns[field].set(cr, self, id, field, vals[field], user, context=rel_context) or []
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/osv/fields.py", line 854, in set
    if inverse_field.ondelete == "cascade":
  File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3150039-161-870617/openerp/fields.py", line 360, in __getattr__
    raise AttributeError(name)
AttributeError: ondelete
```

If I attach the image from the _images_ tab, then no exception occurs, but no image is attached neither.

**Comment by atchuthan** [comment]
@Yajo: already rebased commits from PR#151 to this PR

On Tuesday 05 July 2016 08:18 PM, Yajo wrote:

> This module received a major update in v8 in #151 
> https://github.com/OCA/product-attribute/pull/151, could you merge 
> with your migration to get that please?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> https://github.com/OCA/product-attribute/pull/161#issuecomment-230500261, 
> or mute the thread 
> https://github.com/notifications/unsubscribe/AEc3vnjqV1PVufmfVX_i7ruq4EWcRj8Uks5qSm65gaJpZM4I1CiG.

## 

Thanks & Regards,
Atchuthan

File: product_multi_image/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# © 2014 Serv. Tecnol. Avanzados Pedro M. Baeza
+# © 2015 Antiun Ingeniería S.L. - Jairo Llopis
+# © 2016 Pedro M. Baeza, Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Multiple Images in Products",
+    "version": "9.0.1.0.0",
+    "author": "Serv. Tecnol. Avanzados - Pedro M. Baeza, "
+              "Antiun Ingeniería, Tecnativa, Sodexis, "
```
Comment (yajo) [inline product_multi_image/__openerp__.py:13]: Missing author: https://github.com/OCA/product-attribute/blob/8.0/product_multi_image/__openerp__.py#L11

Comment (atchuthan) [inline product_multi_image/__openerp__.py:13]: @Yajo may I know who I missed?

Comment (yajo) [inline product_multi_image/__openerp__.py:13]: Ah forget it, I saw 1 per line in v8 and did not realize you mixed some in the same line. My fault.

**Comment by yajo** [comment]
I cannot install the `sale` module after `product_multi_image` was installed.

**Comment by yajo** [comment]
Please bear in mind that you probably have a bug that lets dangling images in the table when you uninstall this module. Please refer to https://github.com/OCA/server-tools/pull/485/files#diff-9ebf6925e40a87a77d2472a9134d8a2aR62 and implement that uninstall hook.

I know, that's still in PR, but I'd appreciate help to get it fixed ASAP. Thanks.

**Comment by lasley** [comment]
Hi @SodexisTeam - Are you going to continue with this PR?

**Comment by atchuthan** [comment]
@lasley we were busy with some customer task, so I was unable to work on this.

It seems @Yajo raised an issue on https://github.com/odoo/odoo/issues/13076 and it relates to the issue we are facing now.

and we have to add the uninstall hook as suggested by @Yajo.

We would appreciate if you could submit your changes as PR to this branch.

**Comment by pedrobaeza** [comment]
We are making it, so please wait for our PR. @Yajo, @carlosdauden is the migration already available?

**Comment by yajo** [comment]
Nothing yet AFAIK

**Comment by lasley** [comment]
Thanks for the update guys. This just came up on some of our project scope, so I have some time that can be pushed this way. Sounds like we're on the right track, but let me know if my team can assist in anything. I'll go back to lurker status now 😉

**Comment by pedrobaeza** [comment]
@SodexisTeam, the problems you mentioned were in _base_multi_image_, so basically this module should work right now. Please check.

File: product_multi_image/models/product_product.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    )
+    image_medium = fields.Binary(
+        related='image_main_medium',
+        store=False,
+        multi=False
+    )
+    image_small = fields.Binary(
+        related='image_main_small',
+        store=False,
+        multi=False
```
Comment (yajo) [inline product_multi_image/models/product_product.py]: I'm afraid these `multi` probably don't have the desired effect.
Can you confirm please? Review https://github.com/odoo/odoo/issues/10799#issuecomment-241704243

File: product_multi_image/models/product_product.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        inverse="_inverse_image_ids")
+    image_main = fields.Binary(inverse="_inverse_main_image_large")
+    image_main_medium = fields.Binary(inverse="_inverse_main_image_medium")
+    image_main_small = fields.Binary(inverse="_inverse_main_image_small")
+
+    @api.multi
+    def _inverse_main_image(self, image):
+        for product in self:
+            if image:
+                product.image_ids[0].write({
```
Comment (yajo) [inline product_multi_image/models/product_product.py:30]: 1. Install `product_multi_image`.
2. Install `sale`.

```
2016-09-01 14:01:04,410 20 INFO demo openerp.modules.loading: loading sale/sale_demo.xml
2016-09-01 14:01:17,255 20 ERROR demo openerp.modules.registry: Failed to load registry
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/openerp/modules/registry.py", line 386, in new
    openerp.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/usr/lib/python2.7/site-packages/openerp/modules/loading.py", line 337, in load_modules
    loaded_modules, update_module)
  File "/usr/lib/python2.7/site-packages/openerp/modules/loading.py", line 237, in load_marked_modules
    loaded, processed = load_module_graph(cr, graph, progressdict, report=report, skip_modules=loaded_modules, perform_checks=perform_checks)
  File "/usr/lib/python2.7/site-packages/openerp/modules/loading.py", line 159, in load_module_graph
    _load_data(cr, module_name, idref, mode, kind='demo')
  File "/usr/lib/python2.7/site-packages/openerp/modules/loading.py", line 98, in _load_data
    tools.convert_file(cr, module_name, filename, idref, mode, noupdate, kind, report)
  File "/usr/lib/python2.7/site-packages/openerp/tools/convert.py", line 851, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/usr/lib/python2.7/site-packages/openerp/tools/convert.py", line 934, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/usr/lib/python2.7/site-packages/openerp/tools/convert.py", line 801, in parse
    self.parse(rec, mode)
  File "/usr/lib/python2.7/site-packages/openerp/tools/convert.py", line 804, in parse
    self._tags[rec.tag](self.cr, rec, de, mode=mode)
  File "/usr/lib/python2.7/site-packages/openerp/tools/convert.py", line 708, in _tag_record
    id = self.pool['ir.model.data']._update(cr, self.uid, rec_model, self.module, res, rec_id or False, not self.isnoupdate(data_node), noupdate=self.isnoupdate(data_node), mode=self.mode, context=rec_context )
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/openerp/addons/base/ir/ir_model.py", line 1144, in _update
    res_id = model_obj.create(cr, uid, values, context=context)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/opt/odoo/extra-addons/OCB/product/product.py", line 1192, in create
    product_id = super(product_product, self).create(cr, uid, vals, context=ctx)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 354, in old_api
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/extra-addons/OCB/mail/models/mail_thread.py", line 233, in create
    thread = super(MailThread, self).create(values)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/openerp/models.py", line 4166, in create
    self._fields[key].determine_inverse(record)
  File "/usr/lib/python2.7/site-packages/openerp/fields.py", line 954, in determine_inverse
    self.inverse(records)
  File "/usr/lib/python2.7/site-packages/openerp/fields.py", line 595, in _inverse_related
    other[field.name] = record_value[record]
  File "/usr/lib/python2.7/site-packages/openerp/models.py", line 5761, in __setitem__
    return self._fields[key].__set__(self, value)
  File "/usr/lib/python2.7/site-packages/openerp/fields.py", line 865, in __set__
    record.write({self.name: self.convert_to_write(value)})
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 574, in new_api
    result = method(self._model, cr, uid, self.ids, *args, **old_kwargs)
  File "/opt/odoo/extra-addons/OCB/product/product.py", line 1200, in write
    res = super(product_product, self).write(cr, uid, ids, vals, context=context)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 381, in old_api
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/extra-addons/OCB/mail/models/mail_thread.py", line 279, in write
    result = super(MailThread, self).write(values)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/openerp/models.py", line 3848, in write
    self._fields[key].determine_inverse(self)
  File "/usr/lib/python2.7/site-packages/openerp/fields.py", line 952, in determine_inverse
    getattr(records, self.inverse)()
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/opt/odoo/extra-addons/product-attribute/product_multi_image/models/product_product.py", line 54, in _inverse_main_image_large
    product._inverse_main_image(product.image_main)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/opt/odoo/extra-addons/product-attribute/product_multi_image/models/product_product.py", line 44, in _inverse_main_image
    product.image_ids[0].write({
  File "/usr/lib/python2.7/site-packages/openerp/models.py", line 5756, in __getitem__
    return self._browse(self.env, (self._ids[key],))
ParseError: "tuple index out of range" while parsing /opt/odoo/extra-addons/OCB/sale/sale_demo.xml:319, near
<record id="advance_product_0" model="product.product">
            <field name="name">Deposit</field>
            <field name="categ_id" ref="product.product_category_1"/>
            <field name="type">service</field>
            <field name="list_price">150.0</field>
            <field name="invoice_policy">order</field>
            <field name="standard_price">100.0</field>
            <field name="uom_id" ref="product.product_uom_unit"/>
            <field name="uom_po_id" ref="product.product_uom_unit"/>
            <field name="company_id" eval="[]"/>
            <field name="image" type="base64" file="sale/static/img/advance_product_0-image.jpg"/>
        </record>
```

File: product_multi_image/__init__.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl-3).
+# © 2009 Sharoon Thomas Open Labs Business Solutions
+# © 2014 Serv. Tecnol. Avanzados Pedro M. Baeza
+# © 2015 Antiun Ingeniería S.L. - Jairo Llopis
+# © 2016 Pedro M. Baeza, Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from . import models
+from .hooks import pre_init_hook
```
Comment (yajo) [inline product_multi_image/__init__.py]: Cannot uninstall: `AttributeError: 'module' object has no attribute 'uninstall_hook'`. Import it too please.

**Comment by yajo** [comment]
Please check Travis.

**Comment by pedrobaeza** [comment]
Hi, @SodexisTeam / @atchuthan, any news regarding this PR?

**Comment by lasley** [comment]
I am superseding with #199, please close

**Pull request closed without merging**